primitives/linux-port/column.cpp

```diff
@@ -144,36 +121,102 @@ constexpr getNonEmptyMaskPtrT getNonEmptyMaskPtrTemplate()
   return nullptr;
 }
 
-template<>
+template <>
 constexpr getNonEmptyMaskPtrT getNonEmptyMaskPtrTemplate<1>()
 {
   return getNonEmptyMask1Byte;
 }
 
-template<>
+template <>
 constexpr getNonEmptyMaskPtrT getNonEmptyMaskPtrTemplate<2>()
 {
   return getNonEmptyMask2Byte;
 }
 
-template<>
+template <>
 constexpr getNonEmptyMaskPtrT getNonEmptyMaskPtrTemplate<4>()
 {
   return getNonEmptyMask4Byte;
 }
 
-template<>
+template <>
 constexpr getNonEmptyMaskPtrT getNonEmptyMaskPtrTemplate<8>()
 {
   return getNonEmptyMask8Byte;
 }
 
-template<>
+template <>
 constexpr getNonEmptyMaskPtrT getNonEmptyMaskPtrTemplate<16>()
 {
   return getNonEmptyMask16Byte;
 }
 
+template <ENUM_KIND KIND, typename VT, typename T>
+inline typename VT::MaskType getNonEmptyMaskAux(typename VT::MaskType* nonEmptyMaskAux, uint16_t iter)
+{
+  VT proc;
+  if constexpr (sizeof(T) == sizeof(uint8_t))
+  {
+    return nonEmptyMaskAux[iter];
+  }
+  else if constexpr (sizeof(T) == sizeof(uint16_t))
+  {
+    const char* ptr = reinterpret_cast<const char*>((uint64_t*)nonEmptyMaskAux + iter);
+    return proc.loadFrom(ptr);
+    // return MT{ptr[0], ptr[1], ptr[2], ptr[3], ptr[4], ptr[5], ptr[6], ptr[7]};
+    // return MT{0x0, 0x0};
+  }
+  else if constexpr (sizeof(T) == sizeof(uint32_t) && !std::is_same<T, float>::value)
+  {
+    const char* ptr = reinterpret_cast<const char*>((uint32_t*)nonEmptyMaskAux + iter);
+    return proc.loadFrom(ptr);
```

Comment:
Hey,I think loadFrom will not work right.For uint32_t,only first 32 bits from ptr is useful.We need to turn 0xFF into 0xFFFFFFFF in each element of mask.The loadFrom directly load 128 bits from ptr which causes some problem.

Reply:
Hey.
You are right indeed. I was going to add aux column filtering test first and then actually fix aux column processing.